src/middlewared/middlewared/etc_files/ganesha/ganesha.conf

```diff
@@ -0,0 +1,148 @@
+<%
+    import itertools
+
+    config = middleware.call_sync("nfs.config")
+
+    shares = middleware.call_sync("sharing.nfs.query", [["enabled", "=", True]])
+
+    kerberos_keytabs = middleware.call_sync("datastore.query", "directoryservice.kerberoskeytab")
+
+    gc = middleware.call_sync("datastore.config", "network.globalconfiguration")
+
+    bindip = middleware.call_sync("nfs.bindip", config)
+    sec = middleware.call_sync("nfs.sec", config, kerberos_keytabs)
+
+    export_id = itertools.count(1)
+%>
+
+NFS_CORE_PARAM
+{
+    % if config["mountd_port"]:
+        MNT_Port = ${config["mountd_port"]};
+    % endif;
+    % if config["rpclockd_port"]:
+        NLM_Port = ${config["rpclockd_port"]};
+    % endif;
+
+    % if bindip:
+        Bind_addr = ${bindip[0]};
+    % endif;
+
+    % if config["v4"]:
+        Protocols = 3, 4;
+    % else:
+        Protocols = 3;
+    % endif;
+}
+
+% if config["v4"]:
+    NFSV4
+    {
+        % if config["v4_v3owner"]:
+            Allow_Numeric_Owners = true;
+            Only_Numeric_Owners = true;
+        % endif;
+        % if config["v4_domain"]:
+            DomainName = ${config["v4_domain"]};
+        % endif;
+    }
+
+    % if config["v4_krb_enabled"]:
+        NFS_KRB5
+        {
+            % if gc["gc_hostname_virtual"]:
+                PrincipalName = nfs@${f'{gc["gc_hostname_virtual"]}.{gc["gc_domain"]}'};
+            % else:
+                PrincipalName = nfs@${gc["gc_domain"]};
+            % endif;
+            KeytabPath = /etc/krb5.keytab;
+            Active_krb5 = YES;
+        }
+    % endif;
+% endif;
+
+EXPORT_DEFAULTS
+{
+    % if sec:
+        SecType = ${", ".join(sec)};
+    % endif
+}
+
+% for share in shares:
+    <%
+        clients = share["networks"] + share["hosts"]
+    %>
+
+    % for path in share["paths"]:
+        EXPORT
+        {
+            Export_Id = ${next(export_id)};
+            Path = ${path};
+            Pseudo = ${path};
+
+            % if config["udp"]:
+                Transports = TCP, UDP;
+            % else:
+                Transports = TCP;
+            % endif
+
+            % if clients:
+                Access_Type = None;
+
+                CLIENT
+                {
+                    Clients = ${", ".join(clients)};
+
+                    Access_Type = ${"RO" if share["ro"] else "RW"};
+                }
+            % else:
+                Access_Type = ${"RO" if share["ro"] else "RW"};
+            % endif;
+
+            % if share["mapall_user"]:
+                Squash = AllSquash;
+                <%
+                user = middleware.call_sync("user.query", [("username", "=", share["mapall_user"])])
```

Comment:
Does this account for directory service? Do we need {'extra': {'search_dscache': True}} ?